sdk/tests/example.test.ts

```diff
@@ -0,0 +1,250 @@
+import { Pool, Tick, Position } from '../src/schema'
+import { getLiquidityByY, toPercentage, toPrice } from 'invariant-vara-wasm'
+
+const api = await initGearApi({ providerAddress: Network.Local })
+const admin = await GearKeyring.fromSuri('//Alice')
+
+let token0Address: HexString = null as any
+let token1Address: HexString = null as any
+
+{
+  const token0 = await FungibleToken.deploy(api, admin, 'Coin', 'COIN', 12n)
+  const token1 = await FungibleToken.deploy(api, admin, 'Coin', 'COIN', 12n)
+  await token0.mint(admin.addressRaw, 1000000000000000000000000000000n)
+  await token1.mint(admin.addressRaw, 1000000000000000000000000000000n)
+  token0Address = token0.programId()
+  token1Address = token1.programId()
+}
+
+const INVARIANT_ADDRESS = (await Invariant.deploy(api, admin, 0n)).programId()
+let unsub: Promise<VoidFunction> = null as any
+describe('sdk guide snippets', async function () {
+  this.timeout(80000)
+
+  this.beforeAll(async function () {
+    unsub = subscribeToNewHeads(api)
+  })
+  it('sdk guide', async function () {
+    this.timeout(80000)
+
+    // load invariant contract
+    const invariant = await Invariant.load(api, INVARIANT_ADDRESS)
+
+    // set fee tier, make sure that fee tier with specified parameters exists
+    const feeTier = newFeeTier(toPercentage(1n, 2n), 1n) // fee: 0.01 = 1%, tick spacing: 1
+
+    // If the fee tier does not exist, you have to add it
+    const isAdded = await invariant.feeTierExists(feeTier)
+    if (!isAdded) {
+      await invariant.addFeeTier(admin, feeTier)
+    }
+
+    // set initial price of the pool, we set it to 1.00
+    // all endpoints only accept sqrt price so we need to convert it before passing it
+    const price = toPrice(1n, 0n)
+    const initSqrtPrice = priceToSqrtPrice(price)
+
+    // set pool key, make sure that pool with specified parameters does not exists
+    const poolKey = newPoolKey(token0Address, token1Address, feeTier)
+
+    await invariant.createPool(admin, poolKey, initSqrtPrice)
+
+    // token y has 12 decimals and we want to add 8 actual tokens to our position
+    const tokenYAmount = 8n * 10n ** 12n
+
+    // set lower and upper tick indexes, we want to create position in range [-10, 10]
+    const lowerTickIndex = -10n
+    const upperTickIndex = 10n
+
+    // calculate amount of token x we need to give to create position
+    const { amount: tokenXAmount, l: positionLiquidity } = getLiquidityByY(
+      tokenYAmount,
+      lowerTickIndex,
+      upperTickIndex,
+      initSqrtPrice,
+      true
+    )
+
+    // print amount of token x and y we need to give to create position based on parameters we passed
+    console.log(tokenXAmount, tokenYAmount)
+
+    // load token contracts
+    const tokenX = await FungibleToken.load(api, poolKey.tokenX)
+    const tokenY = await FungibleToken.load(api, poolKey.tokenY)
+
+    // approve transfers of both tokens
+    await tokenX.approve(admin, invariant.programId(), tokenXAmount)
```

Comment:
We slightly changed the structure of interacting with tokens to use a single object that is not related to a specific token address. The token address is then passed as a parameter to the function. It would be good if we had such a structure here as well